spec.bs

```diff
@@ -188,6 +188,9 @@ Each {{Subscriber}} has a <dfn for=Subscriber>subscription controller</dfn>, whi
 
 Each {{Subscriber}} has a <dfn for=Subscriber>active</dfn> boolean, initially true.
 
+Each {{Subscriber}} has an <dfn for=Subscriber>owning Observable</dfn>, an {{Observable}}, which is
```

Comment:
It's actually not necessary for a Subscriber to hold a reference to an Observable. But it is necessary for a (multicast) Observable to hold a reference to the Subscriber, so this is probably okay.

This just caught my eye because with classic "cold" observables, the Observable instance itself could theoretically be GC'ed while the subscription is still alive.

Reply:
I guess you're saying the Subscriber doesn't need to tell the Observable when it is inactive/cleared, and instead the Observable, upon new subscriptions, can just detect if its "active subscriber" is no longer active, and needs to create a new one instead?

Reply:
> The subscriber would probably remove itself as part of its teardown steps, yeah.

That requires a Subscriber->Observable reference though, right? That conflicts with what you said about it being weird to have that reference.

Basically I think it should be possible for Subscriber to not have to have a reference to the Observable it is associated with (so that the Observable can still be GC'd without the Subscriber being killed). Do you agree? In that case when the Subscriber becomes inactive, it just marks itself as inactive, but doesn't touch the Observable at all. The Observable would be responsible for checking its associated Subscriber's "active" status upon future subscriptions. What do you think?

Reply:
As discussed offline:

1. Subscriber does not need to reference Observable at all (upon becoming inactive, it can just set its internal state to inactive, and rely on its associated Observable checking this)
2. Observable doesn't need to strongly reference/_own_ Subscriber. Rather it can just weakly reference its subscriber. Upon subscription, the Observable just checks if (a) the weak subscriber is still alive, and if so, if it's active then add the observer to the active subscriber's collection of observers; otherwise, discard it and start a new subscription (which means running the subscription callback again etc.)

I've made these changes in https://github.com/WICG/observable/pull/197/commits/a819eeb590757e8800d00c07bdf9cd5c163f846e, so I'll resolve this and land the PR.